src/main/java/com/cloudbees/jenkins/support/filter/NameProvider.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.jenkins.support.filter;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Computer;
+import hudson.model.Label;
+import hudson.model.Node;
+import hudson.model.User;
+import hudson.model.View;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Strategy for providing a stream of names to anonymize along with an accompanying name generator.
+ *
+ * @see SensitiveContentFilter
+ * @since TODO
+ */
+@Restricted(NoExternalUse.class)
+public class NameProvider implements ExtensionPoint {
+    private final Supplier<Stream<String>> names;
+    private final Supplier<String> fakes;
+
+    private NameProvider(@Nonnull Supplier<Stream<String>> names, @Nonnull Supplier<String> fakes) {
+        this.names = names;
+        this.fakes = fakes;
+    }
+
+    /**
+     * @return a stream of names to anonymize
+     */
+    public @Nonnull Stream<String> names() {
+        return names.get();
+    }
+
+    /**
+     * @return a new fake name to use for anonymization
+     */
+    public @Nonnull String generateFake() {
+        return fakes.get();
+    }
+
+    /**
+     * @return all registered NameProviders
+     */
+    public static @Nonnull ExtensionList<NameProvider> all() {
+        return ExtensionList.lookup(NameProvider.class);
+    }
+
+    /**
+     * Provides the names of items.
+     */
+    public static final @Extension NameProvider ITEMS = new NameProvider(
+            () -> Jenkins.get()
+                    .getItems().stream()
+                    .flatMap(item -> Arrays.stream(item.getFullName().split("/")))
```

Comment:
I think you should be able to use `getName` instead of `getFullName` once you are using `allItems` since `getFullName` just includes the names of all parents up through the root, and you will include them anyway when using `allIItems`.

However, I think you also need to add `item.getDisplayName`, which might appear in logs and can be distinct from the item's name.

Reply:
Sounds good to me. Do you think `getDisplayName` should be added for any other categories?

Reply:
I don't know if the other ones can be modified to be distinct from `getName`, but subclasses could definitely support it, so it is probably safest to add it for everything that has a `getDisplayName` method.